### PR TITLE
Java DataSource instrumentation.

### LINKED ISSUE #3337: Document auto instrumentation for Java DB queries
Related to https://github.com/getsentry/sentry-java/pull/1095

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
Fixes #3337

### CODE REVIEW

**Comment by maciejwalkowiak** [comment]
Not sure if that's the best place to put these docs - but considering they can be used independently from Spring - and with Spring they can be used either with manual DataSource bean creation or through the 3rd party starter - I think it makes sense to keep it on the separate page.

File: src/platforms/java/common/performance/datasource.mdx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+implementation 'io.sentry:sentry-p6spy:{{ packages.version('sentry.java.spring-boot', '4.4.0') }}'
+```
+
+To create spans around database queries, `P6DataSource` must be configured to use `SentryJdbcEventListener`:
+
+```java
+import io.sentry.p6spy.SentryJdbcEventListener;
+import com.p6spy.engine.spy.P6DataSource;
+
+DataSource dataSource = // original datasource
```
Comment (marandaneto) [inline src/platforms/java/common/performance/datasource.mdx:30]: should we add the import for this?

File: src/platforms/java/common/performance/datasource.mdx [hunk 2]
```diff
@@ -0,0 +1,87 @@
+---
+title: DataSource Instrumentation
+description: "Learn how to instrument javax.sql.DataSource"
+---
+
```
Comment (marandaneto) [inline src/platforms/java/common/performance/datasource.mdx:5]: lets add a note similar to OkHttp this is a preview https://docs.sentry.io/platforms/android/configuration/integrations/okhttp/

**Comment by marandaneto** [comment]
let's keep it as a Draft till cutting a release

File: src/platforms/java/common/performance/datasource.mdx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+This is a preview API, available in `4.4.0-alpha.2`. Features available in a preview API are still in-progress and may have bugs. We recognize the irony. The API may be renamed, changed, or removed in a future version.
+
+</Note>
+
+Sentry provides integration with [P6Spy](https://github.com/p6spy/p6spy) and [DataSource Proxy](https://github.com/ttddyy/datasource-proxy) to intercept *JDBC* database queries and create spans around them.
+
+## P6Spy
+
+To start using *Sentry P6Spy* integration, include following dependency in your project:
```
Comment (imatwawana) [inline src/platforms/java/common/performance/datasource.mdx:16]:
```suggestion
To start using the *Sentry P6Spy* integration, include the following dependency in your project:
```

File: src/platforms/java/common/performance/datasource.mdx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+val dataSource = // original datasource
+val proxyDataSource = ProxyDataSourceBuilder.create(dataSource)
+  .listener(SentryQueryExecutionListener())
+  .build()
+```
+
+## Using with Spring Boot
+
+Spring Boot out of the box does not provide datasource proxying capabilities. We recommend using either *Sentry P6Spy* or *Sentry DataSource Proxy* integration together with a 3rd party library [Spring Boot DataSource Decorator
+](https://github.com/gavlyukovskiy/spring-boot-data-source-decorator). *Sentry Spring Boot Starter* provides auto-configuration classes creating `SentryJdbcEventListener` and `SentryQueryExecutionListener` beans when respective integrations are on the classpath.
```
Comment (imatwawana) [inline src/platforms/java/common/performance/datasource.mdx:95]:
```suggestion
Spring Boot, out of the box, does not provide datasource proxying capabilities. We recommend using either the *Sentry P6Spy* or *Sentry DataSource Proxy* integration with a third party library [Spring Boot DataSource Decorator](https://github.com/gavlyukovskiy/spring-boot-data-source-decorator). *Sentry Spring Boot Starter* provides auto-configuration classes, creating `SentryJdbcEventListener` and `SentryQueryExecutionListener` beans when respective integrations are on the classpath.
```

**Review by imatwawana — APPROVED** [review]
Made some minor wording edits

**Review by marandaneto — APPROVED** [review]
LGTM

**Review by PeloWriter — APPROVED** [review]
LGTM

**Review by bruno-garcia — CHANGES_REQUESTED** [review]
We're having problems publishing the artifacts to central because of this dependency.
And given that the primary requester of this feature is currently not going to use it, we're considering not releasing the new packages. For that reason lets keep this PR in draft state for now please.

**Comment by marandaneto** [comment]
as we deferred publishing the packages, closing it for now

**Pull request closed without merging**